test/read-stream.js

```diff
@@ -54,6 +54,88 @@ describe( 'BlockMap.ReadStream', function() {
 
   })
 
+  it( 'should not close the fd if autoClose = false', function( done ) {
+
+    var filename = path.join( __dirname, '/data/bmap.img' )
+    var blockMap = BlockMap.create( require( './data/version-2.0' ) )
+    var fd = fs.openSync( filename, 'r' )
+    var blockCount = 0
+    var readStream = new BlockMap.ReadStream( null, blockMap, {
+      fd: fd,
+      autoClose: false,
+    })
+
+    readStream
+      .on( 'data', ( block ) => {
+        blockCount++
+        assert.ok( block.address != null, 'block address missing' )
+        assert.ok( block.position != null, 'block position missing' )
+      })
+      .once( 'error', done )
+      .once( 'end', function() {
+        assert.equal( this.blocksRead, blockMap.mappedBlockCount, 'blocksRead mismatch' )
+        assert.equal( this.bytesRead, blockMap.mappedBlockCount * blockMap.blockSize, 'bytesRead mismatch' )
+        assert.equal( this.rangesRead, blockMap.ranges.length, 'rangesRead mismatch' )
+        assert.equal( blockCount, blockMap.mappedBlockCount, 'actual blocks read mismatch' )
+        assert.ok( fs.fstatSync( readStream.fd ) )
+        fs.closeSync( readStream.fd )
+        done()
+      })
+
+  })
+
+  it( 'should start reading at `start`, if specified', function( done ) {
+
+    var filename = path.join( __dirname, '/data/padded-bmap.img' )
+    var blockMap = BlockMap.create( require( './data/version-2.0' ) )
+    var blockCount = 0
+    var readStream = new BlockMap.ReadStream( filename, blockMap, {
+      start: 4096,
+    })
+
+    readStream
+      .on( 'data', ( block ) => {
+        blockCount++
+        assert.ok( block.address != null, 'block address missing' )
+        assert.ok( block.position != null, 'block position missing' )
+      })
+      .once( 'error', done )
+      .once( 'end', function() {
+        assert.equal( this.blocksRead, blockMap.mappedBlockCount, 'blocksRead mismatch' )
+        assert.equal( this.bytesRead, blockMap.mappedBlockCount * blockMap.blockSize, 'bytesRead mismatch' )
+        assert.equal( this.rangesRead, blockMap.ranges.length, 'rangesRead mismatch' )
+        assert.equal( blockCount, blockMap.mappedBlockCount, 'actual blocks read mismatch' )
```

Comment:
> This is node core fs streams' default behavior, I don't think that needs explicit mentioning.

I was just thinking that it might be worth mentioning that the `start` and `end` offsets only apply to the `filename` and don't get 'translated' to the `blockMap` too? \*shrug\*

> Not sure I'm following...

Fair enough, I was just thinking it might be more useful if the unit-tests actually followed expected use-cases more closely (i.e. use `padding-start.img` to test just the `start` offset, use `padding-end.img` to test just the `end` offset, and use `padding-both.img` to test both the `start` and `end` offset together). \*shrug again\*